docs/proposals/oras-attach-multi-platform.md

```diff
@@ -0,0 +1,91 @@
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z" --platform linux/amd64
+```
+
+Resulting image structure:
+
+```console
+demo/alpine:a1a1 (image index) <-- VULNERABLE, EoL attached
+-> demo/alpine:b1b1 (linux/amd64)  <-- VULNERABLE, EoL attached
+-> demo/alpine:c1c1 (linux/arm64)
+```
+
+A new image index is created:
+
+```console
+demo/alpine:z1z1 (image index)
+-> demo/alpine:r1r1 (linux/amd64)
+-> demo/alpine:c1c1 (linux/arm64)
+```
+
+In addition, if a vulnerability is detected and affects all platform-specific images, the parent index and each platform-specific image are patched, generating new digest of each. The outdated multi-platform image and each platform-specific image are marked as invalid using an EoL annotation similar as above.
+
+```console
+demo/alpine:a1a1 (image index)
+-> demo/alpine:b1b1  <-- VULNERABLE
+-> demo/alpine:c1c1  <-- VULNERABLE
+```
+
+Alice has to manually retrieve the new digests and run `oras attach` against each platform-specific image and the parent image index. There is no approach to attach the EoL annotation to the parent image index and propagate to all platform-specific images recursively.
+
+```sh
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z"
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z" --platform linux/amd64
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.artifact.lifecycle" --annotation "vnd.demo.artifact.lifecycle.end-of-life.date=2025-03-20T01:20:30Z" --platform linux/arm64
+```
+
+### Attach a Signature
+
+A DevOps engineer, Bob, wants to attach a cryptographic signature as a referrer to a multi-platform image to ensure integrity and authenticity. Each signature must be attached to both the image index and each platform-specific image to prevent compromise. Given a multi-platform image `demo/alpine:a1a1`, Bob has to run multiple `oras attach` commands:
+
+```sh
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.test.signature" a1a1.sig
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.test.signature" b1b1.sig --platform linux/amd64
+oras attach $registry/demo/alpine:a1a1 --artifact-type "application/vnd.demo.test.signature" c1c1.sig --platform linux/arm64
+```
+
+Resulting image structure:
+
+```console
+demo/alpine:a1a1 (image index) <-- signed with an attached signature a1a1.sig
+-> demo/alpine:b1b1 (linux/amd64)  <-- signed with an attached signature b1b1.sig
+-> demo/alpine:c1c1 (linux/arm64)  <-- signed with an attached signature c1c1.sig
+```
```

Comment:
This scenario is typical as each node is required to be updated individually.

Note that different signature ensures the integrity of different content
1. `a1a1.sig` ensures the integrity of the snapshot of an image bundle.
2. `b1b1.sig` ensures the integrity of the linux/amd64 image.
3. `c1c1.sig` ensures the integrity of the linux/arm64 image.

Reply:
@shizhMSFT  In such assumption, the signature on every node is different from each other because each node is required to be updated individually, correct?